src/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -72,9 +72,12 @@ public OpticalDuplicateFinder(final String readNameRegex, final int opticalDupli
      * unmodified a copy of the list should be passed to this method.
      *
      * @param list a list of reads that are determined to be duplicates of one another
+     * @param keeper a single PhysicalLocation that is the one being kept as non-duplicate, and thus should never be
+     *               annotated as an optical duplicate. May in some cases be null, or a PhysicalLocation not
+     *               contained within the list!
```

Comment:
yes. imagine that you have two PCR copies, A and B and each has optical dups A' and B'. any of these could be keeper, so you want to make sure you do not mark that one as an optical dup.